linkerd/main/src/main/scala/io/buoyant/linkerd/Main.scala

```diff
@@ -95,17 +97,36 @@ object Main extends App {
     server: Server.Initializer,
     name: Path
   ): Closable = {
+    def addrs: Seq[InetSocketAddress] = {
```

Comment:
just a few style nits:
* addrs can be a val and doesn't need a type annotation
* omit unnessary braces for single expression bodies
* omit empty `()` for non-side-effecting methods
* use == for string equality
* that double for-loop can be simplified into a single

something like this

```
    val addrs = if (server.ip.getHostAddress == "0.0.0.0" ) {
      val addresses = for {
        interface <- NetworkInterface.getNetworkInterfaces
        if interface.isUp
        inet <- interface.getInetAddresses
        if !inet.isLoopbackAddress
      } yield new InetSocketAddress(inet.getHostAddress, server.port)
      addresses.toSeq
    } else {
      Seq(server.addr)
    }
```